CSharpLinq/CSharpLinq/Program.cs

```diff
@@ -0,0 +1,186 @@
+﻿using CSharpOOP2;
+using System;
+using System.Collections.Generic;
+//using System.Text.RegularExpressions;
+//using static System.Net.Mime.MediaTypeNames;
+
+#region Task_1
+
+List<dynamic> numbers = new List<dynamic>() { 545, 4, 2, 32, 8, 99, 7, 5, 7, 9 };
+
+var divideby3 = from n in numbers
+                where n % 3 == 0
+                select n;
+
+Console.WriteLine("\u001b[34mConsole output task 1:\u001b[35m");
+PrintCollection(divideby3);
+
+#endregion
+
+#region Task_2
+
+List<dynamic> collection2 = new List<dynamic>() { -7, 9, 11, -15, 13, -9, -10, 12, 14, -8 };
+
+var negativeNumbers = collection2.Where(n => n < 0 || n <= 10);
+
+Console.WriteLine("\n\u001b[34mConsole output task 2:\u001b[35m");
+
+PrintCollection(negativeNumbers);
+
+#endregion
+
+#region Task_3
+
+List<dynamic> collection3 = new List<dynamic>() { "Kitten", "Elephant", "Duckling", "Puppy", "Foal" };
+
+var sortAnimal = from animal in collection3
+                 where animal.Length % 2 == 0
+                 orderby animal ascending
+                 select animal;
+
+Console.WriteLine("\n\u001b[34mConsole output task 3:\u001b[35m");
+
+PrintCollection(sortAnimal);
+
+#endregion
+
+#region Task_4
+
+List<dynamic> cities = new List<dynamic>() { "Minsk", "Warsaw", "Lisbon", "Porto", "Barselona", "Rome" };
+
+var resultCities = cities.Where(city => city[0].Equals('P') && city[city.Length - 1].Equals('o'));
+
+Console.WriteLine("\n\u001b[34mConsole output task 4:\u001b[35m");
+
+PrintCollection(resultCities);
+
+#endregion
+
+#region Task_5
+
+Console.WriteLine("Enter the long string:");
+
+//Lacrimosa dies illa Qua resurget ex favilla Judicandus homo reus
+
+var someStr = Console.ReadLine();
+
+//EXPERIMENT
+//regular Expressions for //"LacrimosaDiesIlla"
+//string[] split = Regex.Split(someStr, @"(?<!^)(?=[A-Z])");
+
+var resultString = someStr.Split(' ').Where(s => char.IsUpper(s[0]));
+
+Console.WriteLine("\n\u001b[34mConsole output task 5:\u001b[35m");
+
+PrintCollection(resultString);
+
+#endregion
+
+#region Task_6
+
+List<dynamic> collections4 = new List<dynamic>() { 7, 9, 11, 15, 13, 9, 9, 12, 14, 9 };
+
+var result4 = collections4.Distinct().OrderByDescending(c => c);
```

Comment:
In task you should have used Reverse() :)

Reply:
Fixed:)